src/connection/directConnectionProvider.js

```diff
@@ -160,7 +166,7 @@ DirectConnectionProvider.prototype.setupEnv = function() {
   if (!that.seleniumConfig.address) {
     // start local driver
     if (that.seleniumConfig.useSeleniumJarFlag) {
-      promises.push(this._downloadBinary(that.binaries['selenium']).then(
+      promises.push(this._getBinaryFileName(BINARIES.SELENIUM).then(
```

Comment:
the name fo the method is misleading as the parameter is not the filename (filename is provided in 'filename' property) but the name of the configuration !

Reply:
for a given configuration name, it gives the path to the executable
(and if it's not found, an archive/executable is downloaded, renamed if needed, and then the new path is returned) 
i agree there is a clash with the 'filename' param, which is the desired name of a downloaded executable (after renaming)